### DISCUSSION TITLE
Cloudlfared

### DISCUSSION THREAD

**Discussion by mrkhachaturov** [okami101/terraform-kube-cluster#2] [Announcements]
Hi @adr1enbe4udou1n , I am trying to deploy apps in my k3s cluster with your article and this repo. Is Cloudflared used for internal purposes? You are using Hetzner LB for external services and Cloudflared for internal ones? Can you also explain how you are providing the bouncer key for CrowdSec in Terraform during provisioning? As I remember we need to retrieve it from a running CrowdSec pod using the CLI.

By the way, when running kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/latest/cert-manager.crds.yam, there is no latest tag available.

**Comment by adr1enbe4udou1n** [comment]
Hi !

Readme updated for cert-manager.
For API Key bouncer from this repo you just have to generate your own key (just a random string) and it should integrates with your traefik without needing of generate it from the crowdsec api pod.

From crowdsec side :

```hcl
set {
    name  = "lapi.env[0].value"
    value = var.bouncer_api_key
}
```

From Traefik bouncer middleware side :

```hcl
    spec = {
      plugin = {
        bouncer = {
          enabled          = true
          crowdseclapikey  = var.bouncer_api_key
          crowdsecLapiHost = "crowdsec-service.crowdsec:8080"
        }
      }
    }
```

As for cloudflared, it's just intended for testing and also for cheaper cluster so i do not use any LB actually. Cloudflared is not a real LB of course, but the k3s internal LB is enough for my personal usage. It ensures additionnally that the cluster cannot be accessed outside of cloudflare, so better for sec.

This project is clearly intended for my personal usage for my cluster and publicly provided as sample. I encourage you to use it as template and adapt it to your own needs.

**Comment by mrkhachaturov** [comment]
@adr1enbe4udou1n , Thanks to you, I mastered Docker Swarm, and now I am mastering Kubernetes. I'm sorry if I often distract you with my questions; I try to find information on my own and not bother you unnecessarily! But when I do ask, you always provide very detailed answers, for which I thank you once again!

Can you please provide example of `crowdsec_whitelistes_config`?

In docker we must mount file to `/etc/crowdsec/parsers/s02-enrich/mywhitelist.yaml`

**Comment by adr1enbe4udou1n** [comment]
Sure, mine is :

```yml
name: crowdsecurity/whitelists
description: "Whitelisted external IPs"
whitelist:
  reason: "Trusted IPs"
  ip:
    - my_public_ip
    - ...IPV4 + IPV6 Hetzner IPs of all kubernetes nodes
```

You just have to pass the file via terraform like so:

```hcl
module "base" {
  source = "github.com/adr1enbe4udou1n/terraform-kube-okami/base"
  crowdsec_whitelists_config = file("${path.module}/configs/crowdsec-whitelists.yaml")
  ...
}
```

The nodes IPs are advised to ensure no internal services are blocked, aka all internal tools that fetch services through Traefik, like Flux which pulls internal gitea instance via the public URL.

**Comment by mrkhachaturov** [comment]
@adr1enbe4udou1n, Hello!

Could you please explain the concept of cloud-native configuration from this repository?

Based on the configuration, you are backing up the cluster to the S3 object store and defining an external cluster with the same object store configuration and credentials.

Does this imply that if you delete the cluster and redeploy it, it will bootstrap as if from an external cluster, which essentially is the backup of the deleted cluster?

I recall you mentioning that this repository is for your personal use. I simply aim to gain a better understanding of the configuration.

Thank you in advance!

**Comment by adr1enbe4udou1n** [comment]
Hello,

Indeed this repo use only the `initdb` bootstrap :

```tf
      bootstrap = {
        initdb = {
          database = var.pgsql_user
          owner    = var.pgsql_user
          secret = {
            name = kubernetes_secret_v1.cluster_auth.metadata[0].name
          }
        }
      }
```

So on every new cluster, it will start with empty data.

As explained in the official docs [here](https://cloudnative-pg.io/documentation/1.23/bootstrap/#the-bootstrap-section), i should use `recovery` mode instead. In this mode, you'll benefit of [PITR](https://cloudnative-pg.io/documentation/current/recovery/#point-in-time-recovery-pitr) recovery.

I confess I didn't work on this for now 😁, but it should be just a matter of terraform parameter for support both initdb and recovery bootstrap.
